src/test/java/uk/ac/ebi/eva/submission/integration/SubmissionWSIntegrationTest.java

```diff
@@ -2,79 +2,122 @@
+        assertThat(Files.isDirectory(Paths.get(submissionDropbox + "/" + submissionId))).isTrue();
+
+        // Test get submission status
+        mvc.perform(get("/v1/submission/" + submissionId + "/status")
+                        .headers(httpHeaders)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpectAll(status().isOk(), content().string(SubmissionStatus.OPEN.toString()));
 
-        mvc.perform(put("/v1/completed")
+        // Test mark submission uploaded
+        mvc.perform(put("/v1/submission/" + submissionId + "/uploaded")
                         .headers(httpHeaders)
-                        .contentType(MediaType.APPLICATION_JSON)
-                        .content(response))
+                        .contentType(MediaType.APPLICATION_JSON))
                 .andExpect(status().isOk());
 
-        SubmissionStatus submissionStatus1 = submissionRepository.findBySubmissionId(response);
-        assertThat(submissionStatus1).isNotNull();
-        assertThat(submissionStatus1.getSubmissionId()).isEqualTo(response);
-        assertThat(submissionStatus.isCompleted()).isEqualTo(true);
+        submission = submissionRepository.findBySubmissionId(submissionId);
+        assertThat(submission).isNotNull();
+        assertThat(submission.getSubmissionId()).isEqualTo(submissionId);
+        assertThat(submission.getStatus()).isEqualTo(SubmissionStatus.UPLOADED.toString());
+        assertThat(submission.getUploadedTime()).isNotNull();
+        assertThat(submission.getCompletionTime()).isNull();
+
+        // Test get submission status
+        mvc.perform(get("/v1/submission/" + submissionId + "/status")
+                        .headers(httpHeaders)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpectAll(status().isOk(), content().string(SubmissionStatus.UPLOADED.toString()));
+
+        // Test mark submission status
+        mvc.perform(put("/v1/submission/" + submissionId + "/status/COMPLETED")
+                        .headers(httpHeaders)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        submission = submissionRepository.findBySubmissionId(submissionId);
+        assertThat(submission).isNotNull();
+        assertThat(submission.getSubmissionId()).isEqualTo(submissionId);
+        assertThat(submission.getStatus()).isEqualTo(SubmissionStatus.COMPLETED.toString());
+        assertThat(submission.getCompletionTime()).isNotNull();
+
+        // Test mark submission status with a wrong status
+        mvc.perform(put("/v1/submission/" + submissionId + "/status/complete")
+                        .headers(httpHeaders)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+
+        // Cleanup - delete directory
+        //new File(submissionDropbox + "/" + submissionId).deleteOnExit();
```

Comment:
Do you still need this, or is it cleaned up automatically elsewhere?

Reply:
Still needed. updated